demos/TicTacToe/Classes/MainMenuScene.h

```diff
@@ -2,18 +2,60 @@
 #define TICTACTOE_DEMO_CLASSES_MAINMENU_SCENE_H_
 
 #include "cocos2d.h"
+#include "firebase/auth.h"
+#include "firebase/database.h"
+#include "firebase/future.h"
+#include "firebase/util.h"
 #include "ui/CocosGUI.h"
+using std::to_string;
+
+void LogMessage(const char*, ...);
+bool ProcessEvents(int);
+void WaitForCompletion(const firebase::FutureBase&, const char*);
+
+std::string GenerateUid(std::size_t);
 
 class MainMenuScene : public cocos2d::Layer, public cocos2d::TextFieldDelegate {
  public:
   // Builds a simple scene that uses the bottom left cordinate point as (0,0)
   // and can have sprites, labels and nodes added onto it.
   static cocos2d::Scene* createScene();
+  virtual void MainMenuScene::update(float);
```

Comment:
All the methods should have a comment describing what they do.

Reply:
[Google Style Guide Functions](https://google.github.io/styleguide/cppguide.html#Function_Comments) , actually all of the comments section is a good read, especially when writing tutorials for other folks to read & learn.